### ISSUE TITLE
Img alt text should be considered when evaluating label text for links 

### ISSUE THREAD

**Issue by kloots** [GoogleChrome/accessibility-developer-tools#156]
The [Link With Unclear Purpose check](https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/audits/LinkWithUnclearPurpose.js) only checks for text content. This means you can get a false negative if a link's content is an image with alt text. Rather than evaluating the text content, the test should be using the [getTextFromDescendantContent util method]( https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/js/Properties.js#L416).

**Comment by jdan** [comment]
Looks like `getTextFromDescendantContent` isn't picking up on alt text. Is there an alternative? Otherwise I'll be checking for `alt` manually.

**Comment by ckundo** [comment]
Looks like this was resolved in #38.

**Comment by ckundo** [comment]
Oops, thought this was resolved but it was actually a PR against tota11y. My mistake!

**Comment by RickSBrown** [comment]
This is a good point and in the same vein as #202.

I'd like to take this further - it would be quite easy to generate a link that was visually meaningful with no content at all, for example using the `content` CSS property.

I'll open a PR for this.

**Comment by RickSBrown** [comment]
You could do the same thing with no content and a CSS background image too.

The PR I have opened allows for this (as long as there is `aria-label` or other text alternative). Should it? 
If it's accessible to ALL users (visual, keyboard, mouse, AT, touch screen etc) then it's all good right?

Even if it's done in a strange way... I could actually imagine wanting to do this when payload size and/or DOM size were of the utmost importance.
